File: 0779-k-th-symbol-in-grammar/0779-k-th-symbol-in-grammar.py
class Solution:
    def kthGrammar(self, n: int, k: int) -> int:
        if n==1:
            return 0
        prevK=self.kthGrammar(n-1,(k+1)//2)
        
        if prevK==0:
            return 0 if k%2==1 else 1
        else:
            return 1 if k%2==1 else 0
       
        

File: 0779-k-th-symbol-in-grammar/0779-k-th-symbol-in-grammar.java
class Solution {
    public int kthGrammar(int n, int k) {
        if(n==1) return 0;
        int prevK=kthGrammar(n-1,(k+1)/2);
        if(prevK==0){
            return k%2==1? 0 : 1;
        }
        else{
            return k%2==1? 1 : 0;
        }
    }
}